### DISCUSSION TITLE
Cryptography error, The cipher's MAC doesn't match the expected value

### DISCUSSION THREAD

**Discussion by swe-lockdrives** [dani-garcia/vaultwarden#5011] [Q&A]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.0
* Web-vault version: v2024.6.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://********************",
  "domain_origin": "*****://********************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "**************",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**************************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "********************",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "off",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": "84861",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

### Vaultwarden Build Version

 1.32.0

### Deployment method

Official Container Image

### Reverse Proxy

Apache2

### Host/Server Operating System

Linux

### Operating System Version

Debian Bookworm

### Clients

Web Vault, Browser Extension, Desktop, Android, iOS

### Client Version

all

### Steps To Reproduce

The first person's Bitwarden app crashed on their cell phone and the problem has been there ever since.

For the second person, the problem just happened until daily use without crash

### Expected Result

The password manager server should work

### Actual Result

For the two people in question, the apps no longer work with the error message mentioned in the title and the Web Vault only shows an infinite loading symbol for the entries/folders

### Screenshots or Videos

![IMG_0930](https://github.com/user-attachments/assets/b97e9a20-d7b5-4396-80ba-3272a9126715)

### Additional Context

![IMG_0930](https://github.com/user-attachments/assets/153799de-7423-468a-922d-dc1e12b774ff)

<img width="760" alt="WEB" src="https://github.com/user-attachments/assets/3b3fbbb4-d21b-4e2d-82c1-3f15aa5783a1">

**Comment by stefan0xC** [comment]
> The first person's Bitwarden app crashed on their cell phone and the problem has been there ever since.

When did this issue first occur? Were they using the iOS beta with `v1.31.0` of Vaultwarden? Could it be related to this issue https://github.com/dani-garcia/vaultwarden/discussions/4767#discussioncomment-10187184?

**Reply by swe-lockdrives** [reply]
I have not used a beta for iOS/Bitwarden. The version was the normal version from the App Store. Fortunately, the people have saved their passwords in organizations so that I can simply delete their user accounts and create new ones. I will report if the error occurs again.

**Comment by maxdallmair** [comment]
We have the same issue on 2 bitwarden instances since yesterday.

**Reply by swe-lockdrives** [reply]
Yes, one person has had it for a few weeks and one person started last Friday. Which version of Vaultwarden were you running when you had the problem? And have you managed to get your data back yet?

**Comment by BlackDex** [comment]
It might be the issue regarding the key per cipher and why Bitwarden disabled that feature.

https://github.com/bitwarden/clients/pull/10621

**Reply by swe-lockdrives** [reply]
If this is the error, then Vaultwarden must have released an update, right?

**Reply by BlackDex** [reply]
Must have an update for what?
Vaultwarden supports this feature already. But if some clients do not or break, that is not something we can fix.

**Reply by swe-lockdrives** [reply]
OK i understand.

Should we now only work via the web version?

It can start with 100 people at any time with one person, then I'm just restoring the backup every day :(

**Comment by JXGA** [comment]
For anyone that follows this, I had to view several threads to get resolved. 

You'll need to restore a backup of your vault. If you don't have a backup, you may not resolve this. 

1. Pull latest (or `V1.32.1`).
2. Stop the server.
3. Replace the `db.sqlite3` file with the backed up file. 
4. Start the server.

If you don't have a backup, you can try to remove the last entry in the `ciphers` table in the sqlite file (sorted by `created_at`). This didn't resolve the issue for me. 

Alternatively, perhaps removing the `user_uuid` and adding a valid `organization_uuid` of the last entry in the ciphers table will be successful.

**Reply by BlackDex** [reply]
It might be simpler maybe. Look for all cipher entries which have the `key` column filled and have an `organization_uuid`. Those are probably the culprit. Would be cool if you could verify this by matching your backups maybe @JXGA ?

**Reply by JXGA** [reply]
@BlackDex The entry I think which was causing the issue was one entered earlier today, it has a `key` but no `organization_uuid`, despite assigning it to an org when adding the entry. 

I tried again to delete just the last entry but the vault remained empty when restarting VW.

I wasn't running the latest version of Vaultwarden (I don't know which version), unsure if this matters.

**Reply by BlackDex** [reply]
There were some issues with the key per cipher. And this looks like that. Might be an older client which added the entry and caused an issue maybe? I have not yet encountered anything like this my self yet, and i do have a few ciphers with an own key.

**Reply by lucascouts** [reply]
Thank you very much

remove the `key` value and replace the `user_uuid` to a valid `organization_uuid` solve the problem, I replace sorted by the latest `updated_at`, because the `cipher` (password item) was created before, I just moved to an organization.

I lost the item that change, but is one for all

**Reply by tekmuscle** [reply]
I've also gotten hit by this and got around it by pulling the latest and clearing out all the `key` values. I had 23 entries with a `key` value going back to August. I have no idea why the issue hit me today. I added a new entry via FF extension and that caused the extension to hang. Then went to ios and got the cipher MAC error. Thank the gods for this thread that pointed me in the right direction. 

edit: I don't know if the 23 entries survived or not. I have a back up with those keys still intact and could recover if I knew what to change. Only one of those entries has an `organization_uuid`, the rest have valid `user_uuid` values. There's no other pattern I see with those entries other than having a value in `key`

**Reply by jack-ben** [reply]
I also had this issue after adding a new user, unfortunately no backup to restore. I remove the newly added user,  shut down the docker, opened the db.sqlite3 DB with sqlite and checked the latest added entries.
`select * from ciphers order by created_at DESC LIMIT 4;`
I deleted the 4 Entries and everything works again. And I straight away did a correct backup :-)
`delete from ciphers order by created_at DESC LIMIT 4;`

**Comment by cl0secall** [comment]
I am able to reproduce this issue using the iOS client with the following procedure:

1. create a new, empty test user
2. create a test entry in testuser's vault
3. Create a new, empty org
4. create  a test entry in test org.
5. create another entry: this will break the user's vault:
  * tap the "+" icon to create an entry
  * fill in the name, username, and generate a password
  * set the owner to the test organization
  * wait approx. 2 minutes
  * after waiting, save the entry
  * the vault immediately becomes inaccessible

The `cipher` table entry for the final test entry will have no `organization_uuid` value set.

Edited to add: I don't need to wait the 2 minutes. I just retested without the wait and it does the same thing.

iOS client version: 2024.9.2 (1106). Vaultwarden server version: 1.31.0

**Reply by stefan0xC** [reply]
Why are you not using the latest Vaultwarden server version?
This should have been fixed by https://github.com/dani-garcia/vaultwarden/pull/4800 (which has been available since [`1.32.0`](https://github.com/dani-garcia/vaultwarden/releases/tag/1.32.0)

**Reply by cl0secall** [reply]
I honestly didn't even look to see if I was on latest. that's entirely my fault. Thanks regardless for the pointer.

**Reply by swe-lockdrives** [reply]
Should we now only work via the web version?

It can start with 100 people at any time with one person, then I'm just restoring the backup every day :(

**Reply by swe-lockdrives** [reply]
> Why are you not using the latest Vaultwarden server version? This should have been fixed by #4800 (which has been available since [`1.32.0`](https://github.com/dani-garcia/vaultwarden/releases/tag/1.32.0)

It said above. It is the client's fault and therefore not fixable on the Server??

**Reply by stefan0xC** [reply]
@swe-lockdrives not sure what you mean? if you look at the PR there's nothing we can do to fix an already corrupted entry if you were using `1.31.0` as explained (because how would we know what entry is missing what `organization_id`?) and so there's nothing to fix serverside anymore.

**Reply by swe-lockdrives** [reply]
Yes, I realize that an already incorrect entry cannot be fixed by a update. My point is that the Bitwarden client cannot create the entry incorrectly.

**Reply by stefan0xC** [reply]
If the client is faulty and sends the organization id in an unrecognizable way (eg. because they for whatever reason change it to `orgID`) I don't think there's anything we can do except fix it when it happens or warn users about using such clients. But this hopefully shouldn't happen with the official Bitwarden clients...

**Reply by swe-lockdrives** [reply]
I had it with 3 users with official Bitwarden clients.

**Reply by stefan0xC** [reply]
With `1.31.0` or does it still happen? Also the clients that caused this were in Beta at the time as far as I know.

**Reply by swe-lockdrives** [reply]
Yes with 1.31.0 and the Bitwarden clients were not in beta, but the normal stable app from the App Store.

**Reply by stefan0xC** [reply]
I meant when we released the fixed version `1.32.0` the client was still in Beta. If users are still running an older version (which they can still do), it's a bit out of our hands.

**Comment by Joshua2504** [comment]
I probably came across the same issue today, running Vaultwarden 1.32.5

I'm sure that this was caused by some entry I added via the latest version of the iOS app which did have an org but no folder/collection assigned which failed, then I selected a collection and was able to save. Sadly I can't reproduce

**Reply by Gryphiusz** [reply]
Same issue here. After trying to save a password to a collection the error occurred.
The web vault seems now empty, tried multiple things. The collection still has all the passwords.

Any way to fix this? Or is restoring a backup the only way?

**Reply by jack-ben** [reply]
Copied from earlier answer:
I also had this issue after adding a new user, unfortunately no backup to restore. I remove the newly added user, shut down the docker, opened the db.sqlite3 DB with sqlite and checked the latest added entries.
`select * from ciphers order by created_at DESC LIMIT 4;`
I deleted the 4 Entries and everything works again. And I straight away did a correct backup :-)
`delete from ciphers order by created_at DESC LIMIT 4;`
